frame/benchmarking/src/lib.rs

```diff
@@ -756,13 +808,20 @@ macro_rules! selected_benchmark {
 macro_rules! impl_benchmark {
 	(
 		{ $( $where_clause:tt )* }
-		NO_INSTANCE $( $name:ident ),*
+		NO_INSTANCE
+		( $( $name:ident ),* )
+		( $( $name_extra:ident ),* )
 	) => {
 		impl<T: Trait> $crate::Benchmarking<$crate::BenchmarkResults> for Module<T>
 			where T: frame_system::Trait, $( $where_clause )*
 		{
-			fn benchmarks() -> Vec<&'static [u8]> {
-				vec![ $( stringify!($name).as_ref() ),* ]
+			fn benchmarks(extra: bool) -> Vec<&'static [u8]> {
+				let mut all = vec![ $( stringify!($name).as_ref() ),* ];
+				if !extra {
+					let extra = vec![ $( stringify!($name_extra).as_ref() ),* ];
```

Comment:
No need for allocation. It can be an array. Optimal solution would be to just return an Iterator from this function but `impl Iterator` is not possible in trait functions :(

```suggestion
					let extra = [ $( stringify!($name_extra).as_ref() ),* ];
```